test/fastmath.jl

```diff
@@ -202,3 +202,10 @@ let a = ones(2,2), b = ones(2,2)
     local c = 0
     @test @fastmath(c |= 1) == 1
 end
+
+struct LitPowTest end
+Base.literal_pow(::typeof(Base.FastMath.pow_fast), ::LitPowTest, ::Val{p}) where {p} = 1
```

Comment:
@musm In most cases, no. There is no need for library authors to add any `literal_pow` methods unless they need to hook into the `literal_pow` mechanism.

Using this PR:
```
julia> struct Test end

julia> Base.:^(::Test, x::Int) = x

julia> t = Test()
Test()

julia> Base.FastMath.pow_fast(::Test, ::Any) = 5

julia> t^3
3

julia> @fastmath t^3
5
```

If you do hook into the `literal_pow` mechanism, then depending on what you're trying to achieve, you may or may not want to add both definitions to indicate what should be done with a literal int power when using `^` with or without  `@fastmath`. It is not required to define both. Continuing the previous example:

```
julia> Base.literal_pow(::typeof(Base.FastMath.pow_fast), ::Test, ::Val{p}) where {p} = 6

julia> @fastmath t^3.3
5

julia> @fastmath t^3
6

julia> x = 3
3

julia> @fastmath t^x
5

julia> t^3
3

julia> t^x
3
```

In this example, `@fastmath` does something special with a literal power, when regular exponentiation does not. You could of course define `Base.literal_pow(::typeof(^), ...)` to make exponentiation to a literal int pow without `@fastmath` behave as you wish.